torch/utils/data/dataset.py

```diff
@@ -259,6 +259,46 @@ def __len__(self):
         return len(self.indices)
 
 
+class ChunkDataset(IterableDataset):
+    r"""
+    Wrapper class for the C++ ChunkDataset class.
+
+    Additional to the role of wrapping its C++ counterpart dataset class,
+    it also allows collation, convertion or transformation through ``collate_fn``.
+    ``ChunkDataset`` extends ``IterableDataset`` because the size of the dataset is
+    unknown.
+
+    Arguments:
+        dataset (Dataset): The whole Dataset
+        collate_fn (optional, callable): Collates, converts or transform batches
+    """
+    def __init__(self, dataset, collate_fn=None):
+        super(ChunkDataset, self).__init__()
+        self.dataset = dataset
+        self.collate_fn = collate_fn
```

Comment:
Why is it the dataset that's handling the collation? This should probably be a functionality separated from this.

Reply:
How does one even create a chunk dataset in Python? We cannot expect people to look into the `torch._C` namespace --- it's private!

Reply:
Regarding the first comment, I agree that using existing `DataLoader` (dataloader.py) `collate_fn` is a cleaner way. The new revision already cotnains that change

Reply:
Regarding the second comment, I have created a public namespace for the chunk dataset. Now the user can `import torch.utils.data.chunk` and get the same functionality

I also created a chunk.rst page which describes the API and give two examples on how to use it. The first one is the simplest way and the second the most advanced scenario.